thunder/torch/__init__.py

```diff
@@ -3907,6 +3913,73 @@ def nll_loss_backward(
     return TensorProxy(like=g, shape=a.shape)
 
 
+@torchsymbol(torch.nn.functional.mse_loss)
+def mse_loss(
+    a: TensorLike,
+    /,
+    target: TensorLike,
+    size_average: None | Any = None,
+    reduce: None | Any = None,
+    reduction: str = "mean",
+) -> TensorLike:
+    utils.check(
+        size_average is None and reduce is None,
+        lambda: f"Deprecated size_average={size_average} and reduce={reduce} is not supported!",
+    )
+    utils.check(
+        reduction in ("none", "sum", "mean"),
+        lambda: f'Expected reduction string to be "none", "sum", or "mean", but it is {reduction}.',
+        exception_type=ValueError,
+    )
+
+    # warn broadcasting
+    if a.size() != target.size():
+        warnings.warn(
+            f"Using a target size {target.size()} that is different to the input size {a.size()}"
+            "This will likely lead to incorrect results due to broadcasting."
+            "Please ensure they have the same size."
+        )
+        a, target = broadcast_tensors(a, target)  # do I need this?
+    out = (a - target) ** 2
```

Comment:
I think your analysis is correct that because `sub` implicitly broadcasts the tensors to a common shape, there does not need to be an explicit broadcast before it